Jenkinsfile

```diff
@@ -29,6 +30,77 @@ pipeline {
                 sh 'build/run make -j\$(nproc) test-integration'
             }
         }
+
+        stage('Integration Tests SetUp') {
+            steps {
+                sh 'tests/scripts/saveImages.sh > BUILD_NUMBER'
+                script{
+                    BUILD_NUM = readFile('BUILD_NUMBER').trim()
+                }
+                echo "Saved images from build ${BUILD_NUM}"
+                stash name: 'repo',excludes: '_output/,vendor/,.cache/,.work/'
+            }
+        }
+
+        stage('Integration Tests - parallel runs') {
+            environment {
+                KUBE_VERSION = "v1.6"
+            }
+            steps {
+                parallel(
+                    "GCE_K8s_v1.7" : {
```

Comment:
can this be run in a loop? there must be a better way to capture a "matrix" in jenkins.